annual-calendar-creator/editorextensions/annual-calendar-creator/src/extension.ts

```diff
@@ -0,0 +1,187 @@
+            "TextVAlign": "middle",
+            ColWidths: Array(TABLE_COLUMNS).fill(CELL_WIDTH),
+            RowHeights: [TITLE_HEIGHT, TITLE_HEIGHT, ...Array(6).fill(CELL_HEIGHT)],
+            "CellFill_0,0": month.headerColor.title,
+            "CellSize_0,0": { "w": TABLE_COLUMNS, "h": 1 },
+            "CellFill_1,0": month.headerColor.days,
+            "CellFill_1,1": month.headerColor.days,
+            "CellFill_1,2": month.headerColor.days,
+            "CellFill_1,3": month.headerColor.days,
+            "CellFill_1,4": month.headerColor.days,
+            "CellFill_1,5": month.headerColor.days,
+            "CellFill_1,6": month.headerColor.days,
+        }
+    });
+
+    table.textAreas.set('Cell_0,0', month.name);
+    table.textStyles.set('Cell_0,0', {
+        [TextMarkupNames.Bold]: true,
+        [TextMarkupNames.Size]: TEXT_FONT_SIZE_MONTH_NAME,
+        [TextMarkupNames.Color]: TEXT_COLOR_PRIMARY,
+        [TextMarkupNames.HAlign]: 'center',
+    });
+
+    for (let i = 0; i < daysOfWeek.length; i++) {
+        let textId = `Cell_1,${i}`;
+        table.textAreas.set(textId, daysOfWeek[i]);
+        table.textStyles.set(textId, {
+            [TextMarkupNames.Bold]: true,
+            [TextMarkupNames.Size]: TEXT_FONT_SIZE_DAY_NAME,
+            [TextMarkupNames.Color]: TEXT_COLOR_PRIMARY,
+            [TextMarkupNames.HAlign]: 'center',
+            [TextMarkupNames.Family]: TEXT_FONT_FAMILY
+        });
+    }
+
+    //dates around current month
+    let firstDayOfCurrentMonth = new Date(year, month.index, 1).getDay();
+    let daysInCurrentMonth = new Date(year, month.index + 1, 0).getDate(); //0-th index gives you the last day of the previous month
+    let endOfLastMonth = new Date(year, month.index, 0).getDate();
+
+    for (let i = 0; i < (daysOfWeek.length * 6); i++) {
+        let textId = `Cell_${2 + Math.floor(i / 7)},${i % 7}`;
+        table.properties.set(`${textId}_VAlign`, 'top');
+
+        let isCurrentMonth = false;
+        let date = 0;
+
+        if (i < firstDayOfCurrentMonth) {
+            date = endOfLastMonth - ((firstDayOfCurrentMonth - 1) - i);
+        } else if (i < (firstDayOfCurrentMonth + daysInCurrentMonth)) {
+            isCurrentMonth = true;
+            date = (i + 1) - firstDayOfCurrentMonth;
+        } else {
+            date = (i + 1) - (firstDayOfCurrentMonth + daysInCurrentMonth);
+        }
+
+        table.textAreas.set(textId, date.toString());
+        table.textStyles.set(textId, {
+            [TextMarkupNames.Size]: TEXT_FONT_SIZE_DAY_NAME,
+            [TextMarkupNames.Color]: isCurrentMonth ? TEXT_COLOR_PRIMARY : TEXT_COLOR_SECONDARY,
+            [TextMarkupNames.HAlign]: 'left',
+            [TextMarkupNames.Family]: TEXT_FONT_FAMILY
+        });
+    }
+}
+
+export function createAnnualCalendar(year: number, months: string[], days: string[]) {
+    let page = viewport.getCurrentPage();
+
+    let MONTHS: Month[] = months.map((name, index) => {
```

Comment:
nit: prefer to have lowercase for variable names and uppercase for constants. could use `rawMonths` snd `months`

Reply:
good point! Updated!